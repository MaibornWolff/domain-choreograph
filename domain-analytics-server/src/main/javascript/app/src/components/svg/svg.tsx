import * as React from 'react';
import { styled } from '~utils/styled';

const Wrapper = styled('div')`
  height: 24px;
  width: 24px;
  svg {
    fill: ${({theme}) => theme.colors.icons.normal.primary};
  }
`;

export interface SvgProps {
  inlineSvg: string;
  className?: string;
}

export const Svg: React.StatelessComponent<SvgProps> = ({inlineSvg, className}) => (
  <Wrapper className={className} dangerouslySetInnerHTML={{__html: inlineSvg}}/>
);
