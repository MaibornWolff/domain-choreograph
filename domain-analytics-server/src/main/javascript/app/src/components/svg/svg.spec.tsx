import { shallow } from 'enzyme';
import * as React from 'react';
import { Svg } from './svg';

describe('Svg', () => {
  it('should render', () => {
    expect(shallow(<Svg inlineSvg={'<svg><text>This is a test</text></svg>'}/>)).toMatchSnapshot();
  });

  it('should render with class name', () => {
    expect(shallow(<Svg
      className="my-class"
      inlineSvg={'<svg><text>This is a test</text></svg>'}
    />)).toMatchSnapshot();
  });
});
