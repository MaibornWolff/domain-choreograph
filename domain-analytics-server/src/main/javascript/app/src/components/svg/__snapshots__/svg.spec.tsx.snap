// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg should render 1`] = `
<styled.div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg><text>This is a test</text></svg>",
    }
  }
/>
`;

exports[`Svg should render with class name 1`] = `
<styled.div
  className="my-class"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg><text>This is a test</text></svg>",
    }
  }
/>
`;
